interfaces/apparmor/backend.go

```diff
@@ -63,6 +70,138 @@ func (b *Backend) Name() interfaces.SecuritySystem {
 	return interfaces.SecurityAppArmor
 }
 
+// Initialize prepares customized apparmor policy for snap-confine.
+func (b *Backend) Initialize() error {
+	// NOTE: It would be nice if we could also generate the profile for
+	// snap-confine executing from the core snap, right here, and not have to
+	// do this in the Setup function below. I sadly don't think this is
+	// possible because snapd must be able to install a new core and only at
+	// that moment generate it.
+	return setupSnapConfineGeneratedPolicy()
+}
+
+// setupSnapConfineGeneratedPolicyImpl inspects the system and sets up local
```

Comment:
That's three levels of indirection, and looking through the code it doesn't look like we need any of it. Can we please have the logic below under the real Initialize, and use the real initialize from tests?

Reply:
Sure, done now. This also removed some extra boiler plate code :)